src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -247,13 +361,21 @@ public void Remove(FilePath fromFilePath, PackageId packageId)
                     LocalizableStrings.CannotFindPackageIdInManifest, packageId));
             }
 
+            if (serializableLocalToolsManifest.Tools == null)
+            {
+                throw new InvalidOperationException(
+                    $"Invalid state {nameof(serializableLocalToolsManifest)} if out of sync with {nameof(toolManifestPackages)}. " +
```

Comment:
I assume because this is not loc'd, this is not intended to be surfaced to users?

Small typo in string:

```suggestion
                    $"Invalid state {nameof(serializableLocalToolsManifest)} is out of sync with {nameof(toolManifestPackages)}. " +
```

Reply:
correct. only dev should see it